examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -358,6 +353,22 @@ void OTAProviderExample::HandleQueryImage(app::CommandHandler * commandObj, cons
             OTAImageHeader header;
             VerifyOrDie(ParseOTAHeader(parser, mOTAFilePath, header) == true);
             VerifyOrDie(sizeof(mSoftwareVersionString) > header.mSoftwareVersionString.size());
+            // Return Image not available if we already updated to the version that's available
```

Comment:
This does not obviously seem related to the polling changes.  Why is it here?

Reply:
This is a bug in the OTA example code. when we test with 3 ota requestors, the first ota requestor successfully updates. we do a query image again on ota 1 and the version is already updated but still with old code we get image available and the ota provider prepares for transfer. now when ota 2 sends a query image it will get busy which is incorrect. The ios OTA provider already does that. we just add that behavior to the ota provider example

Reply:
I am not saying it's not a bug.  I am saying:

1. This chamge has nothing to do with the polling changes.
2. Therefore it should not be in this PR.
3. Moving this part to a separate PR will improve the odds that it will get careful review and make this already-large PR smaller and easier to review.

Reply:
sounds good. will do that

Reply:
done